connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java

```diff
@@ -207,142 +156,77 @@ private SparkSession createSparkSessionWithCatalogs(String... catalogs) {
+            List<String> partitionColumns) throws IOException, ApiException {
         setupTables(
                 SPARK_CATALOG,
                 tableName,
                 DataSourceFormat.PARQUET,
+                location,
                 partitionColumns,
-                isManaged,
-                session);
+                false);
     }
 
     private void setupExternalDeltaTable(
             String catalogName,
             String tableName,
             List<String> partitionColumns,
             SparkSession session) throws IOException, ApiException {
-        setupDeltaTable(catalogName, tableName, partitionColumns, false, session);
-    }
+        String location = generateTableLocation(catalogName, tableName);
+        // The Delta path can't be empty, need to set up here.
+        String partitionClause;
+        if (partitionColumns.isEmpty()) {
+            partitionClause = "";
+        } else {
+            partitionClause = String.format(
+                    " PARTITIONED BY (%s)",
+                    String.join(", ", partitionColumns));
+        }
+        session.sql(String.format("CREATE TABLE delta.`%s`(i INT, s STRING) USING delta", location) + partitionClause);
 
-    private void setupManagedDeltaTable(
-            String catalogName,
-            String tableName,
-            List<String> partitionColumns,
-            SparkSession session) throws IOException, ApiException {
-        setupDeltaTable(catalogName, tableName, partitionColumns, true, session);
+        setupExternalDeltaTable(catalogName, tableName, location, partitionColumns);
     }
 
-    private void setupDeltaTable(
+    private void setupExternalDeltaTable(
             String catalogName,
             String tableName,
-            List<String> partitionColumns,
-            boolean isManaged,
-            SparkSession session) throws IOException, ApiException {
+            String location,
+            List<String> partitionColumns) throws IOException, ApiException {
         setupTables(
                 catalogName,
                 tableName,
                 DataSourceFormat.DELTA,
+                location,
                 partitionColumns,
-                isManaged,
-                session);
+                false);
     }
+
     private void setupTables(
             String catalogName,
             String tableName,
             DataSourceFormat format,
+            String location,
             List<String> partitionColumns,
-            boolean isManaged,
-            SparkSession session) throws IOException, ApiException {
+            boolean isManaged) throws IOException, ApiException {
         Integer partitionIndex1 = partitionColumns.indexOf("i");
         if (partitionIndex1 == -1) partitionIndex1 = null;
         Integer partitionIndex2 = partitionColumns.indexOf("s");
         if (partitionIndex2 == -1) partitionIndex2 = null;
 
-        // TODO: remove this when managed table is officially supported.
```

Comment:
why is this removed.

Reply:
I no longer need to create managed tables to test credentials. I'll add managed table support once UC server is ready.